production/avotes-parser-core/tests/test_encoding_parsing.py

```diff
@@ -0,0 +1,81 @@
+from collections import namedtuple
+
+import pytest
+
+from avotes_parser.core import parse_script, EVMScript
+from avotes_parser.core.ABI.provider import get_cached_etherscan_api
+from avotes_parser.core.decoding import FuncInput, Call
+
+ParsingTestCase = namedtuple(
+    'ParsingTestCase', field_names=['raw_script', 'parsed_script']
+)
+
+parse_with_decoding_examples = (
+    ParsingTestCase(
+        raw_script="0x000000019d4af1ee19dad8857db3a45b0374c81c8a1c632000000044"
+                   "ae962acf00000000000000000000000000000000000000000000000000"
+                   "0000000000000c00000000000000000000000000000000000000000000"
+                   "0000000000000000001d",
+        parsed_script=EVMScript(
+            spec_id='00000001',
+            calls=[
+                Call(
+                    contract_address="0x9d4af1ee19dad8857db"
+                                     "3a45b0374c81c8a1c6320",
+                    function_signature="0xae962acf",
+                    function_name="setNodeOperatorStakingLimit",
+                    inputs=[
+                        FuncInput(
+                            name="_id",
+                            type="uint256",
+                            value=12,
+                        ),
+                        FuncInput(
+                            name="_stakingLimit",
+                            type="uint64",
+                            value=29,
+                        ),
+                    ],
+                    properties=dict(),
+                    outputs=list(),
+                )
+            ]
+        )
+    ),
+)
+
+
+@pytest.fixture(scope='module', params=parse_with_decoding_examples)
+def parse_with_decoding_example(request):
+    """Get positive test case for parsing with encoding of calls."""
+    return request.param.raw_script, request.param.parsed_script
+
+
+@pytest.fixture(scope='module')
+def abi_storage(api_key: str, target_net: str):
+    """Return prepared abi storage."""
+    return get_cached_etherscan_api(api_key=api_key, net=target_net)
+
+
+def test_parse_with_call_decoding(abi_storage, parse_with_decoding_example):
```

Comment:
Could you please provide more test cases:
- multiple calls matching the provided ABI
- single call when ABI is not recognized
- multiple calls when some of the calls are not matched with ABI

It would help us to understand the error-handling approaches better.